addons/adyen_platforms/util.py

```diff
@@ -40,3 +56,44 @@ def __call__(self, request):
         })
 
         return request
+
+def odoo_payments_proxy_control():
+    def decorator(func):
+        @wraps(func)
+        def wrapper(*args, **kw):
+            _logger.debug('Check notification from Odoo Payments')
+
+            adyen_uuid = request.httprequest.headers.get('oe-adyen-uuid')
+            account_id = request.env['adyen.account'].sudo().search([('adyen_uuid', '=', adyen_uuid)])
+            if not account_id:
+                raise Forbidden()
+
+            secret = account_id.proxy_token.encode('utf8')
+            msg_signature = request.httprequest.headers.get('oe-signature')  # base64 encoded hmac sha256 digest
+            msg_timestamp = request.httprequest.headers.get('oe-timestamp')
+
+            if not secret or not msg_signature or not msg_timestamp:
+                raise Forbidden()
+
+            if int(msg_timestamp) + TIMEOUT < int(time.time()):
+                _logger.debug('HTTP request validation failed due to invalid timestamp for route "%s"', request.httprequest.path)
```

Comment:
Prefer `%r` to `"%s"` (for text formatting anyway): %r should take care of escaping inner quotes and stuff, so it's a bit more reliable.